src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java

```diff
@@ -767,27 +770,75 @@ public void testMaxWarnings() throws IOException {
+        antTask.setFile(new File(getPath(FLAWLESS_INPUT)));
 
-    private static class CheckstyleAntTaskLogStub extends CheckstyleAntTask {
+        antTask.execute();
 
-        private final List<String> loggedMessages = new ArrayList<>();
+        final Violation auditStartedMessage = new Violation(1,
+                Definitions.CHECKSTYLE_BUNDLE, "DefaultLogger.auditStarted",
+                null, null,
+                getClass(), null);
+        final Violation auditFinishedMessage = new Violation(1,
+                Definitions.CHECKSTYLE_BUNDLE, "DefaultLogger.auditFinished",
+                null, null,
+                getClass(), null);
 
-        @Override
-        public void log(String msg, int msgLevel) {
-            loggedMessages.add(msg);
+        final List<MessageLevelPair> expectedList = Arrays.asList(
+                new MessageLevelPair("checkstyle version .*", Project.MSG_VERBOSE),
+                new MessageLevelPair("Adding standalone file for audit", Project.MSG_VERBOSE),
+                new MessageLevelPair("To locate the files took \\d+ ms.", Project.MSG_VERBOSE),
+                new MessageLevelPair("Running Checkstyle  on 1 files", Project.MSG_INFO),
+                new MessageLevelPair("Using configuration file:.*", Project.MSG_VERBOSE),
+                new MessageLevelPair(auditStartedMessage.getViolation(), Project.MSG_DEBUG),
+                new MessageLevelPair(auditFinishedMessage.getViolation(), Project.MSG_DEBUG),
+                new MessageLevelPair("To process the files took \\d+ ms.", Project.MSG_VERBOSE),
+                new MessageLevelPair("Total execution took \\d+ ms.", Project.MSG_VERBOSE)
+        );
+
+        final List<MessageLevelPair> loggedMessages = antTask.getLoggedMessages();
+
+        assertWithMessage("Amount of log messages is unexpected")
+                .that(loggedMessages)
+                .hasSize(expectedList.size());
+
+        for (int i = 0; i < expectedList.size(); i++) {
+            final MessageLevelPair expected = expectedList.get(i);
+            final MessageLevelPair actual = loggedMessages.get(i);
+            assertWithMessage("Log messages should match")
+                    .that(actual.getMsg())
+                    .matches(expected.getMsg());
+            assertWithMessage("Log levels should be equal")
+                    .that(actual.getLevel())
+                    .isEqualTo(expected.getLevel());
         }
+    }
 
-        @Override
-        public void log(String msg, Throwable t, int msgLevel) {
-            loggedMessages.add(msg);
+    @Test
+    public void testCheckerException() throws IOException {
+        final CheckstyleAntTask antTask = new CheckstyleAntTaskStub();
+        antTask.setConfig(getPath(CONFIG_FILE));
+        antTask.setProject(new Project());
+        antTask.setFile(new File(""));
+        try {
+            antTask.execute();
+            Assertions.fail("Exception is expected");
```

Comment:
as I mentioned in other PR, assertThrows is a clearer way for checking exceptions

Reply:
done